components/rmrk/Gallery/History.vue

```diff
@@ -197,15 +217,34 @@ export default class History extends mixins(ChainMixin, KeyboardEventsMixin) {
     return this.data.slice(endIndex - this.itemsPerPage, endIndex)
   }
 
-  get uniqType(): string[] {
-    return [...new Map(this.copyTableData.map((v) => [v.Type, v])).keys()]
+  getEventDisplayName(type: HistoryEventType) {
+    return EventToIconMap[type] + ' ' + this.$t(`nft.event.${type}`)
+  }
+
+  get uniqType(): { type: HistoryEventType; value: string }[] {
```

Comment:
Is this just for a case to filter out empty interactions?

Reply:
And I transformed the map to a set just now so that the function will be more readable